### DISCUSSION TITLE
Trouble with Gamepad Input handling

### DISCUSSION THREAD

**Discussion by FlyingChopstick** [dotnet/Silk.NET#537] [Q&A]
Hello!
Following the InputTest (Lab/Experiments), I am trying to connect a xbox 360 gamepad.
Every following function is more or less a copy of the example.

On window Load the input is created, the event attached.
![image](https://user-images.githubusercontent.com/37670423/124311486-82ae0a80-db76-11eb-8f56-1ee7b737ed96.png)

ConnectDevices() passes found gamepads to ConnectGamepad() through ConnectInput()
![image](https://user-images.githubusercontent.com/37670423/124312580-319f1600-db78-11eb-8cb5-962b2b2830cb.png)
![image](https://user-images.githubusercontent.com/37670423/124311315-35ca3400-db76-11eb-8ae0-8be5757461c1.png)

Everything works fine: the window is rendered, it fires all the events. until you disconnect the gamepad. Then it goes to the infinite loop on Window.Run() with Glfw.GetGamepadState() on top of the stack. The window is permanently frozen and the process can't be stopped, except by reboot.
Keyboards and mice are connected the same way, with no problem on connect/disconnect, though the ConnectionChanged event is not fired for them.
I feel like I do something wrong but am unable to find what.
Any help or advice will be appreciated.

Edit: I am using Windows 11 and .NET 5, if it is relevant

**Comment by Perksey** [comment]
> The window is permanently frozen and the process can't be stopped, except by reboot.

This sounds like a Windows 11 bug

> with Glfw.GetGamepadState() on top of the stack

This sounds like a GLFW bug.

We will try to reproduce this on Windows 10.

**Reply by FlyingChopstick** [reply]
Thank you!
